crates/matrix-sdk/src/room/common.rs

```diff
@@ -178,24 +180,47 @@ impl Common {
     /// # });
     /// ```
     pub async fn messages(&self, options: MessagesOptions<'_>) -> Result<Messages> {
-        let request = options.into_request(self.inner.room_id());
+        let room_id = self.inner.room_id();
+        let request = options.into_request(room_id);
         let http_response = self.client.send(request, None).await?;
 
+        #[allow(unused_mut)]
         let mut response = Messages {
             start: http_response.start,
             end: http_response.end,
+            #[cfg(not(feature = "e2e-encryption"))]
+            chunk: http_response
+                .chunk
+                .into_iter()
+                .map(|event| RoomEvent { event, encryption_info: None })
+                .collect(),
+            #[cfg(feature = "e2e-encryption")]
             chunk: Vec::with_capacity(http_response.chunk.len()),
```

Comment:
What's the idea behind these changes to `messages`? It looks like it was already decrypting events before, with less code.

Reply:
The idea is that we don't need to get the olm machine for each single event, which includes locking. I can move it to a method.

Reply:
I see. I guess this is fine then.